libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp

```diff
@@ -376,14 +376,19 @@ bool ESP8266WiFiSTAClass::getAutoReconnect() {
 /**
  * Wait for WiFi connection to reach a result
  * returns the status reached or disconnect if STA is off
- * @return wl_status_t
+ * @return wl_status_t or -1 on timeout
  */
-uint8_t ESP8266WiFiSTAClass::waitForConnectResult() {
+int8_t ESP8266WiFiSTAClass::waitForConnectResult() {
     //1 and 3 have STA enabled
     if((wifi_get_opmode() & 1) == 0) {
         return WL_DISCONNECTED;
     }
+    unsigned long timeout = 60*1000; // number of milliseconds to wait before returning timeout error 

```

Comment:
Please reimplement this in terms of polledTimeout, e.g.:
```
using esp8266::polledTimeout::oneShot;
oneShot timeout(60*1000);
while(!timeout)
{
  if(status() != WL_DISCONNECTED)
    return status();
}
return -1;
```

Reply:
done